File: javascript/sum_from_file.js
'''
Write a method that reads a file of integers, and sums them.
Each integer is on a newline and some begin with "#". Skip those.
'''
const fs = require('fs');

const sumFile = (fileName) => {
  let sum = 0;
  const file = fs.readFileSync(fileName, 'utf8').split('\n');
  file.forEach((num) => {
    if (num[0] !== '#') {
      sum += Number(num)
    }
  })
  return sum;
}

module.exports = sumFile;

File: ruby/sum_from_file.rb
'''
Write a method that reads a file of integers, and sums them.
Each integer is on a newline and some begin with "#". Skip those.
'''
def sum_from_file(filename)
  nums = File.readlines(filename)
    .select { |line| line[0] != "#{}" }
    .map(&:to_i)

  nums.inject(:+)
end
